src/test/regress/sql/functional_deps.sql

```diff
@@ -257,3 +260,145 @@ EXECUTE foo;
 ALTER TABLE articles DROP CONSTRAINT articles_pkey RESTRICT;
 
 EXECUTE foo;  -- fail
+
+-- Known issue - in case there is a column in the target list that is not
+-- explicitly listed in the GROUP BY clause but has a functional dependency on
+-- another column in the GROUP BY clause, Postgres planner might output column
+-- values in grouped rows, that actually cannot be uniquely identified. As it
+-- can introduce instability in the tests, all the queries below are
+-- temporarily tested only with GPORCA.
+-- TODO: remove the hard-coded optimizer option below once this issue is fixed.
+SET optimizer = on;
+
+-- start_ignore
+DROP TABLE IF EXISTS test_table1;
+DROP TABLE IF EXISTS test_table2;
+DROP TABLE IF EXISTS test_table3;
+-- end_ignore
+
+CREATE TABLE test_table1 (a int PRIMARY KEY, b text, c int);
```

Comment:
Does the problem occur without PRIMARY KEY?

Reply:
without PRIMARY KEY you have to list explicitly all columns from target list in the group by clause. If you are asking specificaly about mentioned ADBDEV-4888 - it happens only if columns are not listed explicitly in group by clause.

Reply:
Then it’s probably necessary to somehow reflect in the description that the problem is related only to the PRIMARY KEY.

Reply:
Updated the description

Reply:
I suggest name this PR like
Fix ORCA handling queries with GROUP BY PRIMARY KEY
or so on